core/scripts/chaincli/handler/keeper.go

```diff
@@ -473,27 +539,36 @@ func (k *Keeper) deployUpkeeps(ctx context.Context, registryAddr common.Address,
 
 	var err error
 	var upkeepGetter activeUpkeepGetter
+	var endIdx *big.Int // second arg in getActiveUpkeepIds (breaking change in v2.1)
```

Comment:
this seems to be the 4th arg in getActiveUpkeepIds as shown in `activeUpkeepIds := k.getActiveUpkeepIds(ctx, upkeepGetter, big.NewInt(existingCount), endIdx)`

it took me some time to figure out this is endIdx or count, and this is inclusive or not. Seems we are changing from "count" to "index", you may want to update the function signature of `GetActiveUpkeepIDs(opts *bind.CallOpts, startIndex *big.Int, maxCount *big.Int)` in `type activeUpkeepGetter` as well.

Reply:
sorry for the confusion. I meant 2nd arg on registry function :man_facepalming: .

Reply:
added a bunch of additional comments to hopefully make it less confusing